PurchasesTests/Networking/HTTPClientTests.swift

```diff
@@ -400,10 +400,117 @@ class HTTPClientTests: XCTestCase {
         let systemInfo = RCSystemInfo(platformFlavor: nil, platformFlavorVersion: nil, finishTransactions: false)
         let client = RCHTTPClient(systemInfo: systemInfo)
 
-        client.performRequest("POST", path: path, body: Dictionary.init(),
+        client.performRequest("POST", serially: true, path: path, body: Dictionary.init(),
                                    headers: ["test_header": "value"], completionHandler:nil)
 
         expect(headerPresent).toEventually(equal(true))
     }
+
+    func testPerformSerialRequestPerformsAllRequestsInTheCorrectOrder() {
```

Comment:
tricky thing to test, so here's what I came up with (open to suggestions)
- added a number to each request's body, so that I can use the requestNumber and the call count to verify the calls are done in order
- **test 1:** perform a random number of **serial** requests, ensure that they're all started in order
- **test 2:** perform two **serial** requests that take a while to respond, make sure that they're done in order (sort of duplicate of the first, but I think it's a bit clearer, even if it's not as robust)
- **test 3:** perform two **concurrent** requests that take a while to respond, check that we're not waiting for one to be done before starting the next one. it's the same as test 2 but with concurrent instead of serial.

Reply:
In test 3 they are not taking a while to respond right?

We should also test a combination of what happens if there's a concurrent one while a serial one arrives.

Reply:
in test 3 I hadn't added response times, because I was a bit concerned that it would make tests significantly slower since they're running up to 100 requests. 
I added a very small response time now. 
I'll add tests for the combination of concurrent and serial.